include/envoy/network/connection_handler.h

```diff
@@ -26,6 +26,16 @@ class ConnectionHandler {
    */
   virtual uint64_t numConnections() PURE;
 
+  /**
+   * Increment the return value of numConnections() by one.
+   */
+  virtual void incNumConnections() PURE;
```

Comment:
What is the purpose of this for my understanding? I assume just dealing with the stats value? I will take a look at this in more detail once I better understand the use case. FWIW I have an in-flight PR that also touches this area in terms of creating per-listener/per-worker stats and connection balancing (which in and of itself might be useful as it relates to UDP packet forwarding between workers for QUIC if there is no kernel support).

Reply:
Yes, just for updating stats. They are called by EnvoyQuicDispatcher after parsing part of the packet header or getting a connection close callback from quic stack.

Reply:
OK can you add a TODO here per @alyssawilk to clean this up since as she suspected I'm not thrilled with this. You can assign the TODO to me. I will take a look at this when I get back to my stats/balancing changes since I'm touching similar code and can probably figure out how to deal with this.